app/services/arcgis_api/geocoder.rb

```diff
@@ -48,6 +79,39 @@ def request_headers
     end
 
     def parse_suggestions(response_body)
+      handle_api_errors(response_body)
+
+      response_body['suggestions'].map do |suggestion|
+        Suggestion.new(
+          text: suggestion['text'],
+          magic_key: suggestion['magicKey'],
+        )
+      end
+    end
+
+    def parse_address_candidates(response_body)
+      handle_api_errors(response_body)
+
+      response_body['candidates'].map do |candidate|
+        AddressCandidate.new(
+          address: candidate['address'],
+          location: Location.new(
+            longitude: candidate.dig('location', 'x'),
+            latitude: candidate.dig(
+              'location', 'y'
+            ),
```

Comment:
how come this shorter line had to break?
```suggestion
            latitude: candidate.dig('location', 'y'),
```